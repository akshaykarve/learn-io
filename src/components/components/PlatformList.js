import React from 'react';
import '../ComponentStyle.css';
import Platform from './Platform';

const PlatformList=({platforms,onSelectPlatform})=>{
	return(
		// loop for all platforms
		<>
			<Popup username={props.username}/>
			<div>
				{
					platforms.map((x,i) => {
						return (
							<Platform 
							key={i}
							platform={x}
							name={x.platformName}
							onSelectPlatform={onSelectPlatform}
							/>
						);
					})
				}
			</div>
		</>
	);
}

export default PlatformList;