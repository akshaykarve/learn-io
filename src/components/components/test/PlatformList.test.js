import React from 'react';
import renderer from 'react-test-renderer';
import {default as PlatformList} from "../PlatformList";

describe("Platform List Component", ()=>{
	it('test three platforms', () => {
		
		const tree = renderer.create(
			<PlatformList platforms={[{platformName:"test"},{platformName:"test2"},{platformName:"test3"}]}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('test no platform', () => {
		
		const tree = renderer.create(
			<PlatformList platforms={[]}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('test four platforms', () => {
		
		const tree = renderer.create(
			<PlatformList platforms={[{platformName:"test"},{platformName:"test2"},{platformName:"test3"},{platformName:"test4"}]}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});