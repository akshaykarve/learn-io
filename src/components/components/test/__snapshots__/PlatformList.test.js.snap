// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform List Component test four platforms 1`] = `
<div>
  <div
    className="platformStyle grow"
    onClick={[Function]}
  >
    <img
      alt="platformImage"
      src="https://robohash.org/test?200x200"
    />
    <div>
      <h4>
        test
      </h4>
    </div>
  </div>
  <div
    className="platformStyle grow"
    onClick={[Function]}
  >
    <img
      alt="platformImage"
      src="https://robohash.org/test2?200x200"
    />
    <div>
      <h4>
        test2
      </h4>
    </div>
  </div>
  <div
    className="platformStyle grow"
    onClick={[Function]}
  >
    <img
      alt="platformImage"
      src="https://robohash.org/test3?200x200"
    />
    <div>
      <h4>
        test3
      </h4>
    </div>
  </div>
  <div
    className="platformStyle grow"
    onClick={[Function]}
  >
    <img
      alt="platformImage"
      src="https://robohash.org/test4?200x200"
    />
    <div>
      <h4>
        test4
      </h4>
    </div>
  </div>
</div>
`;

exports[`Platform List Component test no platform 1`] = `<div />`;

exports[`Platform List Component test three platforms 1`] = `
<div>
  <div
    className="platformStyle grow"
    onClick={[Function]}
  >
    <img
      alt="platformImage"
      src="https://robohash.org/test?200x200"
    />
    <div>
      <h4>
        test
      </h4>
    </div>
  </div>
  <div
    className="platformStyle grow"
    onClick={[Function]}
  >
    <img
      alt="platformImage"
      src="https://robohash.org/test2?200x200"
    />
    <div>
      <h4>
        test2
      </h4>
    </div>
  </div>
  <div
    className="platformStyle grow"
    onClick={[Function]}
  >
    <img
      alt="platformImage"
      src="https://robohash.org/test3?200x200"
    />
    <div>
      <h4>
        test3
      </h4>
    </div>
  </div>
</div>
`;
